pkg/quota/v1/evaluator/core/pods_test.go

```diff
@@ -446,3 +448,179 @@ func TestPodEvaluatorUsage(t *testing.T) {
+	fakeClock := clock.NewFakeClock(time.Now())
+	evaluator := NewPodEvaluator(nil, fakeClock)
+	activeDeadlineSeconds := int64(30)
+	testCases := map[string]struct {
+		pod           *api.Pod
+		selectors     []corev1.ScopedResourceSelectorRequirement
+		wantSelectors []corev1.ScopedResourceSelectorRequirement
+	}{
+		"EmptyPod": {
+			pod: &api.Pod{},
+			wantSelectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeNotTerminating},
+				{ScopeName: corev1.ResourceQuotaScopeBestEffort},
+			},
+		},
+		"PriorityClass": {
+			pod: &api.Pod{
+				Spec: api.PodSpec{
+					PriorityClassName: "class1",
+				},
+			},
+			wantSelectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeNotTerminating},
+				{ScopeName: corev1.ResourceQuotaScopeBestEffort},
+				{ScopeName: corev1.ResourceQuotaScopePriorityClass, Operator: corev1.ScopeSelectorOpIn, Values: []string{"class1"}},
+			},
+		},
+		"NotBestEffort": {
+			pod: &api.Pod{
+				Spec: api.PodSpec{
+					Containers: []api.Container{{
+						Resources: api.ResourceRequirements{
+							Requests: api.ResourceList{
+								api.ResourceCPU:                        resource.MustParse("1"),
+								api.ResourceMemory:                     resource.MustParse("50M"),
+								api.ResourceName("example.com/dongle"): resource.MustParse("1"),
+							},
+							Limits: api.ResourceList{
+								api.ResourceCPU:                        resource.MustParse("2"),
+								api.ResourceMemory:                     resource.MustParse("100M"),
+								api.ResourceName("example.com/dongle"): resource.MustParse("1"),
+							},
+						},
+					}},
+				},
+			},
+			wantSelectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeNotTerminating},
+				{ScopeName: corev1.ResourceQuotaScopeNotBestEffort},
+			},
+		},
+		"Terminating": {
+			pod: &api.Pod{
+				Spec: api.PodSpec{
+					ActiveDeadlineSeconds: &activeDeadlineSeconds,
+				},
+			},
+			wantSelectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeTerminating},
+				{ScopeName: corev1.ResourceQuotaScopeBestEffort},
+			},
+		},
+		"OnlyTerminating": {
+			pod: &api.Pod{
+				Spec: api.PodSpec{
+					ActiveDeadlineSeconds: &activeDeadlineSeconds,
+				},
+			},
+			selectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeTerminating},
+			},
+			wantSelectors: []corev1.ScopedResourceSelectorRequirement{
+				{ScopeName: corev1.ResourceQuotaScopeTerminating},
+			},
+		},
+		"CrossNamespaceRequiredAffinity": {
+			pod: &api.Pod{
+				Spec: api.PodSpec{
+					ActiveDeadlineSeconds: &activeDeadlineSeconds,
+					Affinity: &api.Affinity{
+						PodAffinity: &api.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+								{LabelSelector: &metav1.LabelSelector{}, Namespaces: []string{"ns1"}, NamespaceSelector: &metav1.LabelSelector{}},
```

Comment:
test one that only has selectors and one that only has the slice

Reply:
few exist already at the end, but I added a couple for anti-affinity, the number of permutations is large...

Reply:
just for one of required/preferred should be enough. Sorry if I missed it.